src/steps/node.rs

```diff
@@ -32,23 +33,42 @@ impl NPM {
 
     #[cfg(target_os = "linux")]
     fn root(&self) -> Result<PathBuf> {
+        let version = self.version()?;
+        let args = if version < Version::new(8, 11, 0) {
+            ["root", "-g"]
+        } else {
+            ["root", "--location=global"]
+        };
         Command::new(&self.command)
-            .args(&["root", "-g"])
+            .args(args)
             .check_output()
             .map(|s| PathBuf::from(s.trim()))
     }
 
+    fn version(&self) -> Result<Version> {
+        let version_str = Command::new(&self.command)
+            .args(&["--version"])
+            .check_output()
+            .map(|s| s.trim().to_owned());
+        Version::parse(&version_str?).map_err(|err| err.into())
+    }
+
     fn upgrade(&self, run_type: RunType, use_sudo: bool) -> Result<()> {
         print_separator("Node Package Manager");
-
+        let version = self.version()?;
+        let args = if version < Version::new(8, 11, 0) {
+            ["update", "-g"]
+        } else {
+            ["update", "--location=global"]
+        };
         if use_sudo {
             run_type
                 .execute("sudo")
                 .arg(self.pnpm.as_ref().unwrap_or(&self.command))
```

Comment:
Do you know if this will work with pnpm?

Reply:
I've added a check for pnpm in https://github.com/r-darwish/topgrade/pull/949/commits/c9aebd33ca506683f91bacf4a320fd8a553f7993, it should use old arguments.